### PR TITLE
Entire codebase rewritten to use jQuery + small fixes

### PR DESCRIPTION
**Pull request by DJDavid98**
I took the native JS code and made all element selection/manipulation use jQuery instead. This makes the code a lot cleaner in my opinion, and when I found a bug or something that could cause a bug, I tried to correct it. Here's a more detailed list on what I changed specifically:

- For separation of content, I moved all of the JS code to a new file, and I also removed most inline style tags and moved them to JS/CSS
- The huge border blocks were substituted for a single border block with the style set to `outset`, which achieves just about the same effect. If needed, tweak the colors to better fit the old version.
- `#menutooltip` was used multiple times, so I changed it to a class
- The `<hr>`s were hard to follow in the menu markup so I changed them to `<section>`s and added the appropriate styles to make them look like the `<hr>` is still there.
- The shop tooltip for some items was displaying "generates 0 smiles per second" and "You pay Infinity smiles per +1 SPS". I added a check to make sure that this information is only displayed when it makes sense
- When an item was clicked in the shop near the edge of the screen, the mouse tooltip would wrap and display weirdly. I fixed this as well.
- I got rid of the `.padding` element and applied the padding rule directly to #menu
- The `<div>`s with `.button` have been replaced with actual buttons
- Some alignment improvements were made to the Credits modal's floated image
- The redundant loading animations were removed, and the 2nd and 3rd spinner got the animation-delay property, which when given a negative value delay the animation while keeping the element visible
- I also renamed "pinkie.svg" and "octavia.svg" to make the names consistent. _(Originally, I only made the files' first letters uppercase, but my Git client didn't recognize it as a change, so I added the ponies' last names too)_
- Changed self-closing tags to regular tags _(Self-closing is not needed since HTML5)_
- The shop's background color was changed to black
- Added a favicon based on `pinkiehappy.png`

### CODE REVIEW

**Comment by blackhole12** [comment]
* I got rid of the .padding element and applied the padding rule directly to #menu

This will re-create the menu dropping off the end of the page bug (see #19 ), because the height is set to 100% and the padding adds on to this. Yes, HTML is that stupid.

* The redundant loading animations were removed, and the 2nd and 3rd spinner got the animation-delay property, which when given a negative value delay the animation while keeping the element visible

I tried to do this, but it never worked in chrome. Have you tested this?

* Changed self-closing tags to regular tags

I *really hate* not having closed tags. Call me OCD or something, but I think this is the dumbest change HTML ever made. Also, this inevitably makes Visual Studio Code try to close them off improperly. I'm not sure I'm going to accept that change.

* The shop's background color was changed to black

This creates too much contrast. There's a reason it wasn't black before.

* Rewrote using jQuery

I avoid using jQuery unless there's a very compelling reason to use it. In this case, it does seem to make everything much nicer, but i'll have to pick through the code to ensure it's actually worth the change.

* You got rid of doOnLoad()

You *NEED* this function or some browsers will randomly not load things properly, even though HTML says they should.

As much as some of these changes are good, there's way too many regressions in this pull request right now. If you can address the concerns above, I might be able to merge it.

**Comment by DJDavid98** [comment]
Please see my newest commit. Also, I tested everything before making the bull request, the `#menu` overflow did catch me off-guard but I fixed it without the need for a new element.

**Comment by blackhole12** [comment]
This looks better, but I will have to manually merge this for conflict resolution. I'll try to do this when I get back from work later today.

**Comment by DJDavid98** [comment]
I just noticed a bug, please wait for a fix before merging

**Comment by DJDavid98** [comment]
Fixed the issue, should be all good now

**Comment by DJDavid98** [comment]
With everything tested and hopefully nothing broken, I think it's done now. I hope I didn't miss anything that would raise a new issue.

**Comment by blackhole12** [comment]
I am now attempting to merge this into master.

**Comment by blackhole12** [comment]
Comments on merging: Both the jquery and the google-analytics links had their "http:" prefix removed, which immediately broke the request. Why was this change made?

I had to fix a bug where, in the store generation function, you have attr('id','count'+1) instead of attr('id','count'+i), which gave everything the same count ID, and predictably made it very confusing.

You put a function inside a loop in the Generate store HTML function. I moved it out of the loop into it's own function. See: https://jslinterrors.com/dont-make-functions-within-a-loop

Upgrades in the store no longer responded to scrolling. This is because $storeupgrades.offset().top is apparently not actually the same as offsetTop. It appears that offset().top was helpfully attempting to take the scrolling into account... even though that explicitly is not what offset() *means*. < insert gigantic facehoof here>

Upgrades in the menu had a tooltip that stuck at -112px. This is because you referenced count instead of i in the upgradeList loop in updateUpgradesAchievements()

These are all the bugs I could find. I am going to create a /beta/ preproduction environment so we can do more extensive testing.

**Comment by DJDavid98** [comment]
Removing the `http:` is common practice for sites that may use https or http. it makes the URL protocol relative. I've never seen that break any website so far.

As for the function inside the loop, that's called a closure, which is one way to use loop variables inside the `on` handler, which I use in other sections of the code as well (`UpdateStore()`, `OrganizePonies`). See: http://stackoverflow.com/a/750495/1344955

**Comment by blackhole12** [comment]
Removing the http: works only in a web environment. If I am trying to run it off of my local filesystem, it tries to use file:// and the requests fail.

I should note that they need an https:// extension now for anyone using https everywhere.

Visual Studio Code complains about the closure in the function too, so apparently it doesn't understand that.

**Comment by DJDavid98** [comment]
Visual Studio has a tendency to flash big warnings on perfectly okay code. Not the best IDE out there, that's for sure.

**Comment by blackhole12** [comment]
As a side-note, the CSS on the store look awful in Firefox, so I may have to revert the outset/inset border styles.

**Comment by blackhole12** [comment]
I had to do an emergency bugfix to deal with a massive memory leak caused by this fork that was brought to my attention by AClosetBrony on reddit. The +1 clicks were changed to use jQuery, but were still removed using DOM native function, which meant jQuery couldn't do cleanup. The fix is simple:

> function aniEndFunc() {
>   $(this).remove();
> }

Are there any other places we may have forgotten to do jQuery cleanup?
